shell/platform/android/flutter_main.cc

```diff
@@ -37,6 +38,22 @@ extern const intptr_t kPlatformStrongDillSize;
 
 namespace {
 
+// This is only available on API23+, so dynamically look it up.
+// This method is only called once at shell creation.
+// Do this in C++ because the API is avialable at level 23 here, but only 29+ in
+// Java.
+static bool ATrace_isEnabled() {
```

Comment:
If this is already in an anonymous namespace, does it need to be static?

Also consider adding logging in here if `dlopen` or `dlsym` fail.

Should failure to find them be fatal? Should it be fatal in our testing environment?

Reply:
Dlopen would be fatal but dlsym not because we could be on a lower api level

Reply:
Nit: Please call this method name anything other than the name in `libandroid.so` to indicate to the caller that this is not from stock Android. Or to avoid potential issues when if/when the header for this is included in the translation unit.

Or, just follow the style guide for method names here.

Reply:
Nit: Please use `fml::NativeLibrary`. It's probably exactly the same behavior but I'd rather have all interaction with native libraries go through one place in the engine.

Reply:
Renamed it and using NativeLibrary now.

Reply:
(I can't use ResolveSymbol since that doesn't really let you resolve functions)